.github/workflows/ci.yml

```diff
@@ -406,6 +407,43 @@ jobs:
     uses: ./.github/workflows/zeebe-ci.yml
     secrets: inherit
 
+  build-operate-backend:
+    if: needs.detect-changes.outputs.operate-backend-changes == 'true'
+    needs: [detect-changes]
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    env:
+      BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+      - uses: ./.github/actions/setup-build
+        name: Build setup
+        with:
+          java-distribution: adopt
+          maven-cache-key-modifier: operate
+          maven-version: 3.8.6
+          vault-address: ${{ secrets.VAULT_ADDR }}
+          vault-role-id: ${{ secrets.VAULT_ROLE_ID }}
+          vault-secret-id: ${{ secrets.VAULT_SECRET_ID }}
+      - name: Build Maven
+        run: |
+          ./mvnw clean deploy -B -T1C -P -docker,skipFrontendBuild -DskipTests -D skipOptimize -DskipChecks -DaltStagingDirectory=${{ github.workspace }}/staging-${{ env.BRANCH_NAME }} -DskipRemoteStaging=true -Dmaven.deploy.skip=true
+      - name: Build Docker image
+        uses: ./.github/actions/build-platform-docker
+        with:
+          repository: 'registry.camunda.cloud/team-operate/camunda-operate'
+          dockerfile: operate.Dockerfile
```

Comment:
In the operate-merge-ci workflow we didn't build a Docker image into the `team-operate` repository, see https://github.com/camunda/camunda/blob/c2f1e689080bcdc8955413216ec5a56aabefdc40/.github/workflows/operate-merge-ci.yaml#L21. I think we only have that for PR builds via https://github.com/camunda/camunda/blob/c2f1e689080bcdc8955413216ec5a56aabefdc40/.github/workflows/operate-ci.yml#L43 so that a Docker images of a PR branch gets built for testing.

Overall I wonder if we need this job at all in the unified CI. Apparently we don't need the Docker image part (we still need the snapshot build for main branch that is further down below). And my impression is that the Maven build is also covered by other jobs, like the java-checks one. I'm not fully certain about that as I struggle a bit with the overall complexity, but maybe we could discuss that?

We could need the job later when we add more of the actual Operate CI into the unified CI, i.e. the tests that get run via https://github.com/camunda/camunda/blob/c2f1e689080bcdc8955413216ec5a56aabefdc40/.github/workflows/operate-ci.yml.

Reply:
I think the Operate merge CI still builds the image locally (and thus verifies as part of the merge queue that it is buildable), it just does not [push](https://github.com/camunda/camunda/blob/c2f1e689080bcdc8955413216ec5a56aabefdc40/.github/workflows/operate-merge-ci.yaml#L21) it.

That is actually the same behavior as this PR since the [push input](https://github.com/camunda/camunda/blob/b50c5d56eea76f8d19b2814d35ffede66c42226f/.github/actions/build-platform-docker/action.yml#L26) is `false` by default, meaning build but no push. @kellervater explicitly passing `push: false` is redundant in https://github.com/camunda/camunda/pull/24522/commits/dea03610de5c81cb89594b3cd00ef844b62353de

Reply:
> But it seems to cause confusion when left out.

Then lets add a comment, I'd prefer to not overwrite default parameters.

Reply:
Thanks Christian, I agree with you reasoning about the Docker image. Let's keep it then.